wolfcrypt/src/aes.c

```diff
@@ -13635,32 +13635,49 @@ static WARN_UNUSED_RESULT int S2V(
     }
     if (ret == 0)
 #endif
-    {
+
+    if ((numAssoc > 126) || ((nonceSz > 0) && (numAssoc > 125))) {
+        /* See RFC 5297 Section 7. */
+        WOLFSSL_MSG("Maximum number of ADs (including the nonce) for AES SIV is"
+                    " 126.");
+        ret = BAD_FUNC_ARG;
+    }
+
+    if (ret == 0) {
         XMEMSET(tmp[1], 0, AES_BLOCK_SIZE);
         XMEMSET(tmp[2], 0, AES_BLOCK_SIZE);
 
         ret = wc_AesCmacGenerate(tmp[0], &macSz, tmp[1], AES_BLOCK_SIZE,
                                  key, keySz);
-        if (ret == 0) {
-            ShiftAndXorRb(tmp[1], tmp[0]);
-            ret = wc_AesCmacGenerate(tmp[0], &macSz, assoc, assocSz, key,
-                                     keySz);
-            if (ret == 0) {
-                xorbuf(tmp[1], tmp[0], AES_BLOCK_SIZE);
-            }
-        }
     }
 
     if (ret == 0) {
-        if (nonceSz > 0) {
-            ShiftAndXorRb(tmp[0], tmp[1]);
-            ret = wc_AesCmacGenerate(tmp[1], &macSz, nonce, nonceSz, key,
-                                     keySz);
+        /* Loop over authenticated associated data AD1..ADn */
+        byte tmpi = 0;
+        for(word32 ai = 0; ai < numAssoc; ++ai) {
```

Comment:
For portability please declare `word32 ai` on line above. Code consistency please use `for (` add space.